src/main/java/com/zufar/onlinestore/security/jwt/filter/JwtAuthenticationProvider.java

```diff
@@ -3,64 +3,58 @@
 import com.zufar.onlinestore.security.exception.JwtTokenException;
 import com.zufar.onlinestore.security.jwt.JwtClaimExtractor;
 import com.zufar.onlinestore.security.jwt.JwtTokenFromAuthHeaderExtractor;
-import com.zufar.onlinestore.security.jwt.JwtTokenValidator;
-
+import jakarta.servlet.http.HttpServletRequest;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
 import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
-import org.springframework.security.core.context.SecurityContextHolder;
 import org.springframework.security.core.userdetails.UserDetails;
 import org.springframework.security.core.userdetails.UserDetailsService;
 import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
 import org.springframework.stereotype.Service;
 
+import java.util.Date;
 import java.util.Optional;
 
-import jakarta.servlet.http.HttpServletRequest;
-import lombok.RequiredArgsConstructor;
-import lombok.extern.slf4j.Slf4j;
-
 @Slf4j
 @Service
 @RequiredArgsConstructor
 public class JwtAuthenticationProvider {
 	private final JwtTokenFromAuthHeaderExtractor jwtTokenFromAuthHeaderExtractor;
-	private final JwtTokenValidator jwtTokenValidator;
 	private final JwtClaimExtractor jwtClaimExtractor;
 	private final UserDetailsService userDetailsService;
 
 	public Optional<UsernamePasswordAuthenticationToken> get(final HttpServletRequest httpRequest) {
-		try {
+		Optional<String> jwtTokenOptional = jwtTokenFromAuthHeaderExtractor.extract(httpRequest);
+		if (jwtTokenOptional.isEmpty()) {
+			return Optional.empty();
+		}
 
-			//Try to retrieve token from Authentication Header of HttpServletRequest
-			Optional<String> jwtTokenOptional = jwtTokenFromAuthHeaderExtractor.extract(httpRequest);
-			if (jwtTokenOptional.isEmpty()) {
-				return Optional.empty();
-			}
+		final String jwtToken = jwtTokenOptional.get();
 
-			//Validate jwtToken
-			final String jwtToken = jwtTokenOptional.get();
-			if (!jwtTokenValidator.isValid(jwtToken)) {
-				return Optional.empty();
+		try {
+			Date expirationDate = jwtClaimExtractor.extractExpiration(jwtToken);
+			Date now = new Date();
+			if (expirationDate.before(now)) {
+				throw new JwtTokenException("Jwt token is expired");
 			}
 
-			//Try to retrieve userEmail from jwtToken
-			final String userEmail = jwtClaimExtractor.extractUsername(jwtToken);
-			if (userEmail == null || SecurityContextHolder.getContext().getAuthentication() != null) {
-				return Optional.empty();
+			final String userEmail = jwtClaimExtractor.extractEmail(jwtToken);
+			if (userEmail == null) {
```

Comment:
What would happen if email will be initialized with empty? Should we handle it too?